trade-finance-manager-ui/scripts/moj-filter-hide.ts

```diff
@@ -0,0 +1,52 @@
+import $ from 'jquery';
+
+declare global {
+  interface Window {
+    MOJFrontend: any;
+  }
+}
+
+type FilterVisibility = 'SHOWN' | 'HIDDEN';
+
+const filterSelector = '.moj-filter-layout__filter';
+const toggleButtonContainerSelector = '.moj-action-bar__filter';
+const toggleButtonContainer = document.querySelector(toggleButtonContainerSelector)!;
+
+const params = (toggleButtonContainer as HTMLElement).dataset;
+const FILTER_VISIBILITY_SESSION_KEY = `${params.filteredTableId}-filters-visibility-state`;
+
+const setFilterVisibility = (visibility: FilterVisibility) => {
+  sessionStorage.setItem(FILTER_VISIBILITY_SESSION_KEY, visibility);
+};
+
+const cachedShownStatus = sessionStorage.getItem(FILTER_VISIBILITY_SESSION_KEY) as FilterVisibility | null;
+
+if (cachedShownStatus) {
+  setFilterVisibility(cachedShownStatus);
+}
+
+// Initially display filters, but hide them if they were previously hidden
+const startHidden = cachedShownStatus === 'HIDDEN';
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-call, no-new, @typescript-eslint/no-unsafe-member-access
```

Comment:
Can we remove such eslint rules please

Maybe it'd be best to have this as a JS file instead, like the other client side scripts

Reply:
@abhi-markan I'd quite like to keep new files to typescript if possible for maintainability, is this one of the places where we might want to look at disabling some of the eslint rules on files within the script folders you mentioned before rather than having the eslint-disables in line?

Reply:
I'd also prefer TS

Please see [the comment below](https://github.com/UK-Export-Finance/dtfs2/pull/3526/files#r1763142121) - disabling these eslint rules for this line only would also be required in JS and is what the current JS implementation in portal (`portal/scripts/moj-filters.js`) also has to do